src/server/adminJs/tabs/donationTab.ts

```diff
@@ -265,6 +269,55 @@ export const buildDonationsQuery = (
   return query;
 };
 
+export const FillPricesForDonationsWithoutPrice = async () => {
+  const donationsWithoutPrice = await Donation.createQueryBuilder('donation')
+    .where('donation.valueUsd IS NULL')
+    .orderBy({ 'donation.createdAt': 'DESC' })
+    .getMany();
+  const coingeckoAdapter = new CoingeckoPriceAdapter();
+  const filledPrices: { donationId: number; price: number }[] = [];
+  logger.debug(
+    'FillPricesForDonationsWithoutPrice donation ids',
+    donationsWithoutPrice.map(d => d.id),
+  );
+  for (const donation of donationsWithoutPrice) {
+    let price: number;
+    try {
+      const token = await Token.findOneBy({ symbol: donation.currency });
+      if (!token || !token.coingeckoId) continue;
+      price = await coingeckoAdapter.getTokenPriceAtDate({
+        date: moment(donation.createdAt).format('DD-MM-YYYY'),
+        symbol: token.coingeckoId,
+      });
+    } catch {
+      return {
+        redirectUrl: '/admin/resources/Donation',
+        record: {},
+        notice: {
+          message: `${donationsWithoutPrice.length} donations without price found. ${filledPrices.length} prices filled`,
+          type: 'success',
+        },
+      };
+    }
+    donation.valueUsd = donation.amount * price;
+    donation.priceUsd = price;
+    filledPrices.push({ donationId: donation.id, price });
+    await donation.save();
+  }
```

Comment:
I think we are missing to update the projects totals, but that might be a little heavy. What do you think? @RamRamez

Reply:
Yeah, great point, thanks @CarlosQ96.
I added to update project totals, user total received and user total donated.